pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -605,7 +602,7 @@ def _update_best_and_save(
             del_list.append(delpath)
 
         # do not save nan, replace with +/- inf
-        if torch.isnan(current):
+        if isinstance(current, torch.Tensor) and torch.isnan(current):
             current = torch.tensor(float('inf' if self.mode == "min" else '-inf'))
```

Comment:
should this always be on cpu? or should it be on `current.device`?

Reply:
Great question ! I am not sure. What do you think ?

Reply:
IMO it should live on current.device, since all the other tensors (especially current if not nan) also live on this device.